tests/ttnn/unit_tests/operations/test_max_pool2d.py

```diff
@@ -289,3 +289,149 @@ def test_pool_core_nondivis(
+):
+    in_n, in_c, in_h, in_w = act_shape
+    kernel_h, kernel_w = kernel_size
+    pad_h, pad_w = padding
+    stride_h, stride_w = stride
+    dilation_h, dilation_w = dilation
+
+    if 2 * pad_h > kernel_h or 2 * pad_w > kernel_w:
+        pytest.skip("Invalid case")
+
+    if (kernel_h == 3 and pad_h != 1) or (kernel_h == 2 and pad_h != 0):
+        pytest.skip("kernel size and padding combination not supported")
+
+    out_h = math.floor((in_h + 2 * pad_h - (dilation_h * kernel_h - 1) - 1) / stride_h) + 1
+    out_w = math.floor((in_w + 2 * pad_w - (dilation_w * kernel_w - 1) - 1) / stride_w) + 1
+    if out_w % nblocks != 0:
+        pytest.skip(f"Unsupported case when out_w ({out_w}) % nblocks ({nblocks}) != 0")
+
+    if in_c % 16 != 0:
+        pytest.skip("Current maxpool writer needs nchannels to be multiple of 16!")
+
+    if in_c == 16 and dtype == ttnn.bfloat8_b and in_n * in_h * in_w > 600000:
+        pytest.skip("This case runs out of memory on Grayskull")
+
+    if in_n >= 16 and in_c >= 64 and dtype == ttnn.bfloat8_b and is_wormhole_b0():
+        pytest.skip("This case runs out of memory on Wormhole b0")
+
+    if (
+        is_wormhole_b0()
+        and act_shape == [16, 64, 112, 112]
+        and kernel_size == (3, 3)
+        and padding == (1, 1)
+        and stride == (2, 2)
+        and dilation == (1, 1)
+        and dtype == ttnn.bfloat16
+    ):
+        pytest.skip("Issue #6992: Statically allocated circular buffers in program clash with L1 buffers on core range")
+
+    torch.manual_seed(0)
+    torch.set_printoptions(precision=3, sci_mode=False, linewidth=500, threshold=10000, edgeitems=32)
+
+    ## construct the tensor in NCHW shape
+    act = torch.randn(act_shape, dtype=torch.bfloat16)
+    # act = torch.zeros(act_shape, dtype=torch.bfloat16)
+    # act = torch.ones(act_shape, dtype=torch.bfloat16)
+    # act = torch.arange(0, volume(act_shape), dtype=torch.bfloat16).reshape(act_shape)
+    # for n in range(act_shape[0]):
+    #     for c in range(act_shape[1]):
+    #         for h in range(act_shape[2]):
+    #             for w in range(act_shape[3]):
+    #                 act[n, c, h, w] = 1 + n + h + w + c # + torch.rand(1) * 0.15
+    # torch.save(act, "act.pt")
+    # act = torch.load("act.pt")
+
+    ## this op expects input tensor as { N, 1, H * W, C }, so rearrange and reshape tensor
+    ## but before that, make sure in_c is multiple of tile width
+    act_shape = (in_n, 1, in_h * in_w, in_c)
+    act_permuted = torch.permute(act, (0, 2, 3, 1))
+    act_reshaped = act_permuted.reshape(act_shape)
+
+    reader_patterns_cache = {}
+    max_pool = ttnn.MaxPool2d(
+        kernel_size=(kernel_h, kernel_w),
+        stride=(stride_h, stride_w),
+        padding=(pad_h, pad_w),
+        dilation=(dilation_h, dilation_w),
+        dtype=dtype,
+        device=device,
+        batch_size=in_n,
+        input_height=in_h,
+        input_width=in_w,
+        reader_patterns_cache=reader_patterns_cache,
```

Comment:
Use the new version `ttnn.max_pool2d_new`

Reply:
Done.